### ISSUE TITLE
how to make the tree more intelligent for further splits

### ISSUE THREAD

**Issue by chloe-wang** [microsoft/LightGBM#3337]
## Summary

sometimes, we have a feature which is not informative for single split, because there is no information gain. However, it may make the further splits easier if we split it. For example, we have a  dataset, which has the house price as the  label,  and has features like city name, distance to the center of the city, the location etc. City name is not informative for single split if the average house price is the same for different cities. However, if we split it into different cities, it may make the location split much easier as different locations in different cities are informative. Of course, the easy way is to combine the city name and location into one single feature. But if we have thousands of features,  the feature dimension will explode.

**Comment by btrotta** [comment]
This is a known limitation of this type of model since the tree-building algorithm is "greedy": it makes the most loss-reducing split at each node but does not look ahead further than this. If we instead tried to do a global optimization on the tree structure (i.e. choose the best overall tree structure, by considering all possible sequences of splits) the running time of the algorithm would increase exponentially. 

Other types of models (e.g. neural networks) do not have this limitation, but the trade-offs are longer training time and more risk of overfitting.

**Comment by chloe-wang** [comment]
@btrotta thanks for the reply. Yes, we understood that this is the limitation of the tree-building algorithm.  Just wondering that whether there are some tricks to overcome this issue to some extent, like some kind of feature engineering.

**Comment by btrotta** [comment]
I don't know of any such tricks unfortunately, apart from creating combined features as you mentioned above.

**Comment by chloe-wang** [comment]
@btrotta thanks a lot.

**Comment by StrikerRUS** [comment]
Closing as this is general question and not directly related to LightGBM.
